server/src/main/java/org/opensearch/indices/pollingingest/DefaultStreamPoller.java

```diff
@@ -188,6 +197,7 @@ protected void startPoll() {
                         continue;
                     }
                     blockingQueue.put(result);
+                    lastSuccessfulPointer = result.getPointer();
```

Comment:
does `lastSuccessfulPointer ` need to be a class field? looks like it can be a var for this method `startPoll `